Kernel/Syscalls/sched.cpp

```diff
@@ -29,6 +30,21 @@ KResultOr<int> Process::sys$donate(pid_t tid)
     return 0;
 }
 
+KResultOr<int> Process::sys$donate_peer(int fd)
+{
+    REQUIRE_PROMISE(stdio);
```

Comment:
The description claims only sockets are supported, should this be `REQUIRE_PROMISE_FOR_SOCKET_DOMAIN(...)` or even directly `REQUIRE_PROMISE(unix)`?

Also we don't seem to validate that the description is a socket anywhere?